xayn-ai/src/ltr/list_net/mod.rs

```diff
@@ -1341,4 +1793,24 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn test_serialize_deserialize_list_net() {
+        let list_net = ListNet::new_with_random_weights();
+        let mut buffer = Vec::new();
+        list_net.clone().serialize_into(&mut buffer).unwrap();
+        let list_net2 = ListNet::deserialize_from(&*buffer).unwrap();
+        assert_approx_eq!(f32, list_net.dense1.weights(), list_net2.dense1.weights());
+        assert_approx_eq!(f32, list_net.dense1.bias(), list_net2.dense1.bias());
+        assert_approx_eq!(f32, list_net.dense2.weights(), list_net2.dense2.weights());
+        assert_approx_eq!(f32, list_net.dense2.bias(), list_net2.dense2.bias());
+        assert_approx_eq!(f32, list_net.scores.weights(), list_net2.scores.weights());
+        assert_approx_eq!(f32, list_net.scores.bias(), list_net2.scores.bias());
+        assert_approx_eq!(
+            f32,
+            list_net.prob_dist.weights(),
+            list_net2.prob_dist.weights()
+        );
+        assert_approx_eq!(f32, list_net.prob_dist.bias(), list_net2.prob_dist.bias());
+    }
```

Comment:
This file si pretty huge, can we move the tests to a directory `tests` here and have something like `tests/ltr`, `tests/training`, or some other split.

Reply:
In this or another pr.

Reply:
Sure, I should be able to put it in this PR.